### DISCUSSION TITLE
Can you alias relational fields? For example when using junction collections or M2O relations.

### DISCUSSION THREAD

**Discussion by DanielBiegler** [directus/directus#11097] [Archive]
Hi there, I was trying to tidy up my API requests/responses to be more dense, because I have junction collections and many-to-one relations. Meaning my example response currently is:

```json
{
    "id_m2o": {
        "id_foo": { "name": "example" },
        "id_bar": { "name": "example" },
        "id_baz": { "name": "example" }
    }
}
```

Ideally I'd like it to be 

```json
{
    "foo": "example",
    "bar": "example",
    "baz": "example"
}
```

From a usability perspective I'd love to just tell the SDK to use the following options:

```javascript
{
    fields: [ /*...*/ ],
    filter: { /*...*/ },
    alias: { foo: "id_m2o.id_foo.name" },
}
```

But I got the Error:

```json
{"errors":[{"message":"\"alias\" key/value can't contain a period character `.`","extensions":{"code":"INVALID_QUERY"}}]}
```

So now I'm unsure if Directus supports this? I tried looking for other discussions or issues but the closest thing I could find was #3633 which is similar but not quite the same.

**Comment by joselcvarela** [comment]
Just made a PR to document this https://github.com/directus/directus/pull/11108/files
The alias on SDK is the one used on API side, so I am afraid this is not possible to do
